app/controllers/api/status_controller.rb

```diff
@@ -3,11 +3,32 @@ class Api::StatusController < Api::ApplicationController
 
   def check
     if params[:projects].any?
+      # Try to get all the projects passed in.
       @projects = params[:projects].group_by{|project| project[:platform] }.map do |platform, projects|
         projects.each_slice(1000).map do |slice|
           Project.platform(platform).where(name: slice.map{|project| project[:name] }.uniq).includes(:repository, :versions)
         end.flatten.compact
       end.flatten.compact
+
+      # If there are any that don't get found, downcase them and try again
+      if params[:projects].length != @projects.length
+        # "slugify" the platform projects that we found
+        found = @projects.map { |project| slug_project project }
+        # Find any projects that were passed that we didn't find, to try them again but case insensitive
+        missing = params[:projects].select { |project| !found.include?(slug_project project) }
+        lowercase_project_named = missing.group_by{|project| project[:platform] }.map do |platform, projects|
+          projects.each_slice(1000).map do |slice|
+            slice.map do |project|
```

Comment:
You're not actually taking advantage of the slicing since you're doing a query per project... can we lowercase all of the project names and then do a single query per slice instead?

Reply:
and maybe do like a `lower(name) in (?)` where `?` is a list of names?